src/main/java/io/supertokens/webserver/api/thirdparty/UserAPI.java

```diff
@@ -70,22 +70,20 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IO
             UserInfo user = null;
             if (userId != null) {
                 io.supertokens.pluginInterface.useridmapping.UserIdMapping userIdMapping = UserIdMapping
-                        .getUserIdMapping(this.getTenantIdentifier(req), main, userId,
-                                UserIdType.ANY);
+                        .getUserIdMapping(this.getTenantIdentifierStorageFromRequest(req).toAppIdentifier(), userId, UserIdType.ANY);
```

Comment:
why using this function here and not from the storageLayer? Please be consistent - search for all places in which you are using this function directly.